pkg/cmd/roachtest/tests/restore.go

```diff
@@ -660,6 +662,217 @@ func registerRestore(r registry.Registry) {
 			require.NotEqual(t, 3, maxPauses, "the job should have paused at least once")
 		},
 	})
+
+	durationGauge := r.PromFactory().NewGaugeVec(prometheus.GaugeOpts{Namespace: registry.
```

Comment:
for each new metric the roachtest writer creates, I think we'd also want to apply the following labels: `cluster_id`, `host`, `cloud`, `release_branch` etc. I _think_ these can be applied via the prom server config, correct? If not, I'll probably need to construct an API around the `PromFactory()` so the roachtest write doesn't have to worry about that stuff.

Reply:
`host` is part of the prom config [1]; we could easily add `cloud`; `cluster_id`, `release_branch` and other test-specific labels could be added by the roachtest (test_runner) via (cloud) metadata. E.g., prometheus config. currently exports `test_name` metadata label,

```
- source_labels: [__meta_gce_metadata_test_name]
    separator: ;
    regex: (.*)
    target_label: test_name
    replacement: ${1}
    action: replace
```

[1] https://grafana.testeng.crdb.io/prometheus/targets